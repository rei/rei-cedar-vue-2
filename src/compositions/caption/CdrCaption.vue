<template>
  <div class="cdr-caption cdr-inset--squish">
    <p
      v-if="summary"
      class="cdr-caption__summary">
      {{ summary }}
    </p>
    <cite
      v-if="credit"
      class="cdr-text cdr-text--citation">
      {{ credit }}
    </cite>
  </div>
</template>

<script>

/**
 * Cedar 2 component for captions
 * @version 0.0.1
 * @author [REI Software Engineering](https://rei.github.io/rei-cedar/)
 */
export default {
  name: 'CdrCaption',
  props: {
    /** Caption summary text */
    summary: String,
    /** Caption credit text */
    credit: String,
  },
};
</script>

<style>
  @import '../../css/settings/_index.pcss';
  @import './styles/CdrCaption.pcss';
</style>
