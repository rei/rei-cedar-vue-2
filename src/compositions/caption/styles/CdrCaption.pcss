/* ==========================================================================
  # CdrCaption
  
  All values should map to variables in
  vars/CdrCaption.vars.pcss
  in order to allow for theming

  TOC:

    :Base - CdrCaption
========================================================================== */

.cdr-caption {
  max-width: 530px;

  &__summary {
    @include redwood-display-20;

    margin-bottom: 0;

    & + .cdr-caption__cite {
      display: inline-block;

      /* TODO: replace with spacing token */
      margin-top: 8px;
    }
  }
}
