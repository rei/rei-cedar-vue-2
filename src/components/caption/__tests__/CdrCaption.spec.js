import { mount } from '@vue/test-utils';
import CdrCaption from 'componentdir/caption/CdrCaption';

describe('CdrCaption', () => {
  it('renders correctly', () => {
    const wrapper = mount(CdrCaption, {
      propsData: {
        summary: 'This is a summary',
        credit: 'This is a credit',
      }
    });
    expect(wrapper.element).toMatchSnapshot();
  });
});
