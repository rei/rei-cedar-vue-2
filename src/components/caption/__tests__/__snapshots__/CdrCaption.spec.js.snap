// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CdrCaption renders correctly 1`] = `
<div
  class="cdr-caption cdr-space-inset-one-x-squish"
>
  <p
    class="cdr-caption__summary"
  >
    This is a summary
  </p>
  <cite
    class="cdr-text cdr-text--citation"
  >
    This is a credit
  </cite>
</div>
`;
